### PR TITLE
feat(Text, Flex, Box): typed html attributes

### PR DESCRIPTION
**Pull request by IsaevAlexandr**
- fully types DOM attributed depends on `as` props for `Text`, `Box` and `Flex` components with backward compatibility;
- added tests for `Text` component;

related to https://github.com/gravity-ui/uikit/issues/1581

### CODE REVIEW

**Comment by chelentos** [comment]
I doubt the correctness of using of `Box` for `Text`. Semantically, these components are needed for different tasks. For example, does the `Text` component need props for its width/height and spacing?

**Comment by IsaevAlexandr** [comment]
> I doubt the correctness of using of `Box` for `Text`. Semantically, these components are needed for different tasks. For example, does the `Text` component need props for its width/height and spacing?

these are reasonable remarks. The use of the `Box` component in this example is necessary in order to work correctly with basic props such as `qa`, `styles`, `className` and `ref` and other. Like base logic fore thouse examples. Out of the box you get the ability to use Text component something like this:
```tsx
<>
  <Text spacing={{mr: 2}}>Some text 1</Text>
  <Text>Some text 1</Text>
<>
```

**Comment by chelentos** [comment]
> Out of the box you get the ability to use Text component something like this:
> 
> ```tsx
> <>
>   <Text spacing={{mr: 2}}>Some text 1</Text>
>   <Text>Some text 1</Text>
> <>
> ```

I think the question then might be why are we adding this behavior for `Text` only?

**Comment by IsaevAlexandr** [comment]
> > Out of the box you get the ability to use Text component something like this:
> > ```tsx
> > <>
> >   <Text spacing={{mr: 2}}>Some text 1</Text>
> >   <Text>Some text 1</Text>
> > <>
> > ```
> 
> I think the question then might be why are we adding this behavior for `Text` only?

`Text` is one more in a row. `Flex`, `Card` and, i hope, it will be more components some what extends from `Box`

File: src/components/Text/__stories__/Text.stories.tsx [hunk 1]
```diff
@@ -101,3 +103,45 @@ export const WordBreak: Story = {
         wordBreak: 'break-all',
     },
 };
+
+const LabelWithControlledFocusStory = () => {
+    const ref = React.useRef<HTMLLabelElement>(null); // don't delete this ref - needed to check currect react html type inshurance
```
Comment (korvin89) [inline src/components/Text/__stories__/Text.stories.tsx:108]: currect -> correct

**Review by amje — APPROVED** [review]

**Review by korvin89 — APPROVED** [review]

**Pull request merged**